packages/pluggableWidgets/app-events-native/src/AppEvents.tsx

```diff
@@ -18,61 +18,62 @@ export class AppEvents extends Component<Props> {
     private lastOnOnline = 0;
     private lastOnOffline = 0;
     private onLoadTriggered = false;
-    private timeoutHandle?: any;
+    private onTimeoutAction: ScheduledExecution | undefined;
     private unsubscribeNetworkEventListener?: NetInfoSubscription;
 
     async componentDidMount(): Promise<void> {
-        if (this.props.onResumeAction) {
-            AppState.addEventListener("change", this.onAppStateChangeHandler);
-        }
-
-        if (this.props.onTimeoutAction) {
-            const schedule = this.props.timerType === "once" ? setTimeout : setInterval;
-            this.timeoutHandle = schedule(() => executeAction(this.props.onTimeoutAction), this.props.delayTime * 1000);
-        }
-
-        if (this.props.onOnlineAction || this.props.onOfflineAction) {
-            this.isConnected = (await NetInfo.fetch()).isConnected;
-            this.unsubscribeNetworkEventListener = NetInfo.addEventListener(this.onConnectionChangeHandler);
-        }
-
-        if (!this.onLoadTriggered && this.props.onLoadAction?.canExecute) {
-            this.onLoadTriggered = true;
-            executeAction(this.props.onLoadAction);
-        }
+        AppState.addEventListener("change", this.onAppStateChangeHandler);
+        this.registerScheduledExecution();
+        await this.registerNetworkEvents();
+        this.triggerOnLoadEvent();
     }
 
     componentWillUnmount(): void {
-        if (this.props.onUnloadAction && this.props.onUnloadAction.canExecute) {
-            executeAction(this.props.onUnloadAction);
-        }
+        AppState.removeEventListener("change", this.onAppStateChangeHandler);
+        this.onTimeoutAction?.cancel();
+        this.unRegisterNetworkEvents();
+        this.triggerOnUnloadEvent();
+    }
 
+    render(): null {
+        return null;
+    }
+
+    private async onResume(): Promise<void> {
         if (this.props.onResumeAction) {
-            AppState.removeEventListener("change", this.onAppStateChangeHandler);
+            executeAction(this.props.onResumeAction);
         }
+        this.onTimeoutAction?.schedule();
+        this.lastOnResume = Date.now();
+        await this.registerNetworkEvents();
+    }
 
-        if (this.unsubscribeNetworkEventListener) {
-            this.unsubscribeNetworkEventListener();
-        }
+    private onBackground(): void {
+        this.onTimeoutAction?.cancel();
+        this.unRegisterNetworkEvents();
+        this.triggerOnUnloadEvent();
+    }
 
-        if (this.props.onTimeoutAction && this.timeoutHandle != null) {
-            const clear = this.props.timerType === "once" ? clearTimeout : clearInterval;
-            clear(this.timeoutHandle);
+    private async registerNetworkEvents(): Promise<void> {
+        if (this.props.onOnlineAction || this.props.onOfflineAction) {
+            this.isConnected = (await NetInfo.fetch()).isConnected;
```

Comment:
What if this promise fails?

Reply:
So it "maybe" an option for an api saying isConnected? :)

Reply:
Shall we add a retry mechanism? 
What would be the correct number of retries?
Would we be blocking the NF till done?

Reply:
I can't clearly derive if the NetInfo api will even fail and obviously as it returns a promise the assumption can always hold true.

Reply:
Should the onload/onresume event even be blocked while this thing retries or loads?

Reply:
@tamelbrun